### DISCUSSION TITLE
Negative samples

### DISCUSSION THREAD

**Discussion by OzKmbatWmbat** [AlexeyAB/darknet#8373] [Q&A] [answered]
So I'm currently trying to improve the datasets I am creating, I just stumbled across the Negative sampling.

What I am wanting to know is what is better putting the negative samples in the train or valid files separately ( only in train or only in valid) or put them in both.

From my thought process it would be better in both as it would need to have "valid " negatives samples to work off and ones to also train itself on, but would make sense to have it split in either one or the other as well as having it in valid would like I said give it valid negatives, but having them solely in train would have images it was training on so it will learn to ignore that way. 

If anyone can shed some light on this this would be brilliant.

Cheers

**Comment by stephanecharette** [comment, marked as answer]
See:  https://www.ccoderun.ca/programming/darknet_faq/#train_and_valid

**Reply by OzKmbatWmbat** [reply]
Cheers mate, that has some real good info in it.
